### ISSUE TITLE
"Share access" always displays every role, regardless of the user's permissions.

### ISSUE THREAD

**Issue by dwrensha** [sandstorm-io/sandstorm#2407]
We should hide roles that are beyond the current user's permission level.

**Comment by alexander-vidoni** [comment]
I agree. This looks like a user could grant higher permissions than he owns when re-sharing the grain.
It also tells a visitor more about the role system than he/she should know about it.

**Comment by paulproteus** [comment]
Adding `customer-request` because I'm answering a support ticket about this right now.

**Comment by paulproteus** [comment]
Being more specific about the labels. h/t @kentonv
